src/lints/struct_marked_non_exhaustive_changed_type.ron

```diff
@@ -0,0 +1,74 @@
+SemverQuery(
+    id: "struct_marked_non_exhaustive_changed_type",
```

Comment:
Unfortunately this name is misleading. The other `marked_non_exhaustive` lints are about *adding* `#[non_exhaustive]` to a type. The "marked" is the action that caused the breakage.

This name seems to follow the pattern until at the last second it adds "changed type" which will be confusing to our users.

How about naming this lint `non_exhaustive_struct_changed_type`? That follows the precedent of `constructible_struct_changed_type` and other "<optional adjective> <subject> <action causing breakage>" lints?